packages/components/organisms/f-registration/test-utils/component-objects/f-registration.component.js

```diff
@@ -1,66 +1,130 @@
- * @param {String} userInfo.firstName The user's first name
- * @param {String} userInfo.lastName The user's last name
- * @param {String} userInfo.email The user's e-mail address
- * @param {String} userInfo.password The user's password
- */
-exports.submitRegistrationForm = (userInfo) => {
-    exports.waitForRegistrationForm();
-    firstNameInput().setValue(userInfo.firstName);
-    lastNameInput().setValue(userInfo.lastName);
-    emailInput().setValue(userInfo.email);
-    passwordInput().setValue(userInfo.password);
-    createAccountButton().click();
-};
+    isInputFieldDisplayed(fieldName){
+        return this.fields[fieldName].input.isDisplayedInViewport();
+    };
+    /**
+     * @description
+     * Inputs user details into the registration component and submits the form.
+     *
+     * @param {Object} userInfo
+     * @param {String} userInfo.firstName The user's first name
+     * @param {String} userInfo.lastName The user's last name
+     * @param {String} userInfo.email The user's e-mail address
+     * @param {String} userInfo.password The user's password
+     */
+    submitForm(userInfo){
+        this.fields.firstName.input.setValue(userInfo.firstName); 
+        this.fields.lastName.input.setValue(userInfo.lastName);
+        this.fields.email.input.setValue(userInfo.email);
+        this.fields.password.input.setValue(userInfo.password);
+        this.createAccountButton.click();
+    };
 
-exports.waitForRegistrationForm = () => {
-    registrationComponent().waitForExist();
-};
+    countOfErrorsDisplayed(){
+        return this.errors.length;
+    };
+
+    isEmptyErrorDisplayed(fieldName){
+        return this.fields[fieldName].emptyError.isDisplayedInViewport();
+    };
 
-exports.isFirstNameEmptyErrorDisplayed = () => firstNameEmptyError().isDisplayed();
-exports.isFirstNameMaxLengthErrorDisplayed = () => firstNameMaxLengthError().isDisplayed();
-exports.isFirstNameInvalidErrorDisplayed = () => firstNameInvalidError().isDisplayed();
+    isMaxLengthErrorDisplayed(fieldName) {
+        return this.fields[fieldName].maxLengthError.isDisplayedInViewport();
+    };
 
-exports.isLastNameEmptyErrorDisplayed = () => lastNameEmptyError().isDisplayed();
-exports.isLastNameMaxLengthErrorDisplayed = () => lastNameMaxLengthError().isDisplayed();
-exports.isLastNameInvalidErrorDisplayed = () => lastNameInvalidError().isDisplayed();
+    isEmailExistsErrorDisplayed(){
+        return this.fields.email.existsError.isDisplayedInViewport();
+    }
 
-exports.isEmailEmptyErrorDisplayed = () => emailEmptyError().isDisplayed();
-exports.isEmailInvalidErrorDisplayed = () => emailInvalidError().isDisplayed();
-exports.isEmailExistsErrorDisplayed = () => emailExistsError().isDisplayed();
-exports.isPasswordEmptyErrorDisplayed = () => passwordEmptyError().isDisplayed();
+    isInvalidErrorDisplayed(fieldName) {
+        return this.fields[fieldName].invalidError.isDisplayedInViewport();
+    };
+    
+    displayRequestErrorText(){
```

Comment:
Perhaps rename this to `getRequestErrorText`?

Reply:
yep, makes sense, will do!